test/rules/ordered-imports/groups-complex/test.ts.lint

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env node
+
+// comment outside of imports
+import {app_f} from 'app/foo';
+import {x} from '@pkg/foo';
+~~~~~~~~~~~~~~~~~~~~~~~~~~~ [Import sources of different groups must be sorted by: extra, ^app, ^@pkg, current dir, parent_dir .]
+~~~~~~~~~~~~~~~~~~~~~~~~~~~ [Import sources within a group must be alphabetized.]
+
+import {app_b} from 'app/bar';
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ [Import sources with same group order must be grouped together.]
```

Comment:
@adidahiya I think your observation here is key.  From the user's perspective, the first time there is a block out of order it is out of order.  It doesn't matter if that is because the group it should be in has already been seen or if the group they are adding isn't the next group up.  In both cases it is simply out of order.

Once I started reviewing how the code works and what is presented to the user I added a pretty fundamental change the completely removes the "must be grouped together" failure check.  Instead the code now simply looks to see if groups are in order and if they aren't it adds a failure at the first block that is out of order and stops there.

This not only makes the logic in the code much more clear but as you observed this makes the user experience better.

The idea is that once the first block of imports is found that has an import that is out of order, then all future blocks also have blocks out of order.  So instead of flagging all blocks we just flag the first block that is causing the issue and provide a fix that will fix all blocks.

This makes the case for manually fixing the import ordering much more clear because people can solve one item at a time.  After they fix the first item that is out of order, then they move on to fixing the next one and so on.  It also helps to keep the autofix code more simple because the first time we see a failure it returns a fix for the entire set of import blocks.

I put this change on it's own commit so you can easily review the changes this causes for the linting output.  Let me know if you think I should make it work a different way.  I think this works very well, but maybe you have a better idea to try.